### PR TITLE
VideoBackends: support multiple compute images for some backends

### PR DESCRIPTION
**Pull request by iwubcode**
This adds multi compute texture support for a few backends (D3D, Vulkan, OGL).  I couldn't figure out D3D12's descriptor stuff and Metal also was a bit odd.

This is pulled from #10362 and updated to the current class structure (with a few fixes to validation layers and re-testing GPU decoding).

### CODE REVIEW

File: Source/Core/VideoBackends/D3D/D3DState.cpp [hunk 1]
```diff
@@ -242,6 +243,25 @@ void StateManager::SetComputeShader(ID3D11ComputeShader* shader)
...
+    }
+    else
+    {
+      D3D::context->OMSetRenderTargets(m_pending.framebuffer->GetNumRTVs(), rtv_null.data(),
+                                       nullptr);
+    }
+    m_dirtyFlags |= DirtyFlag_Framebuffer;
+    m_last_compute_rtv_count = m_pending.framebuffer->GetNumRTVs();
+  }
+
```
Comment (TellowKrinkle) [inline Source/Core/VideoBackends/D3D/D3DState.cpp:264]: Is there a reason you need to unbind everything to run a compute shader?

Comment (iwubcode) [reply Source/Core/VideoBackends/D3D/D3DState.cpp:264]: Now that you mention it, this is unnecessary.  Not sure why I initially had it.  I did see an error in validation errors:

> The Shader Resource View dimension declared in the shader code (TEXTURE2DARRAY) does not match the view type bound to slot 0 of the Pixel Shader unit (BUFFER)

I notice that our lovely bounding box has type buffer.  I'm not sure why my changes would impact error that though.  I need to test master.  Regardless, seems there's a flaw somewhere.  I will investigate.

Comment (iwubcode) [reply Source/Core/VideoBackends/D3D/D3DState.cpp]: @TellowKrinkle - thanks for looking this over.  I took this out, the same validation error happens on master with gpu texture decoding on.  We should resolve it but it at least doesn't impact this review!

**Comment by iwubcode** [comment]
Build errors don't seem related, complaint is:

> ../Source/Core/Common/Version.cpp:8:10: fatal error: Common/scmrev.h: No such file or directory
    8 | #include "Common/scmrev.h"

**Comment by AdmiralCurtiss** [comment]
Looks like the first build after switching back from my PR to the old scmrev system fails, though I have no idea why.

**Review by TellowKrinkle — APPROVED** [review]

**Pull request merged**